controllers/common/clusterState.go

```diff
@@ -124,12 +124,9 @@ func (i *ClusterState) readGrafanaServiceAccount(ctx context.Context, cr *v1alph
 }
 
 func (i *ClusterState) readGrafanaConfig(ctx context.Context, cr *v1alpha1.Grafana, client client.Client) error {
-	currentState, err := model.GrafanaConfig(cr)
-	if err != nil {
-		return err
-	}
+	currentState := model.GrafanaConfig(cr)
```

Comment:
Just want some clarification on this, the reason for deleting the `err` here is that it doesn't actually provide any value to us except returning it, which it will anyway since it errors out?  What's the specific lint rule for this? i'd like to check it out locally with my linter.

Reply:
The lint says the following:
```
controllers/model/grafanaConfig.go:12:58: GrafanaConfig - result 1 (error) is always nil (unparam)
```

and also:
```
func GrafanaConfig(cr *v1alpha1.Grafana) (*v1.ConfigMap, error) {
                                                         ^ess of a variable cannot be nil (staticcheck)
```

So it's both https://github.com/mvdan/unparam and https://staticcheck.io/ that finds the error but in different ways.